src/components/chart-preview/ChartPreview.tsx

```diff
@@ -52,6 +55,42 @@ export const ActualChart = ({
     useSaveCsvData(csv, "chart-data");
   };
 
+  const onFigureDownloadClick = (e: any) => {
+    //TODO rework getting parent node
+    const fullFigureNode =
+      e.target.parentNode.parentNode.parentNode.parentNode.parentNode;
```

Comment:
Have you tested that this is always a consistent relationship across different chart types? Just wondered if different types might have different nesting.